### ISSUE TITLE
[java] A false positive about MissingStaticMethodInNonInstantiatableClass

### ISSUE THREAD

**Issue by 777777clearlove777777** [pmd/pmd#4531]
**Affects PMD Version:7.0.0-rc1** 

**Rule: MissingStaticMethodInNonInstantiatableClass**

Please provide the rule name and a link to the rule documentation:
<https://docs.pmd-code.org/latest/pmd_rules_java_errorprone.html#missingstaticmethodinnoninstantiatableclass>

**Description:**
This rule can report a warning when a class is annotated by `@UtilityClass`, hence PMD should report a warning at line 4 because class `Bar2` has this annotation. However, class `Bar1` at line 2 has no annotations and PMD still reports a warning. 

**Code Sample demonstrating the issue:**

```java
import lombok.experimental.UtilityClass;
class Bar1 {}    //  report a warning, but this is a FP
@UtilityClass
class Bar2 {}    //  report a warning
```

**Expected outcome:**

PMD reports a violation at line 2, but that's wrong. That's a false positive.
